File: app/src/main/java/com/victoria/bleled/util/arch/base/BaseBindingActivity.java
package com.victoria.bleled.util.arch.base;

import androidx.databinding.DataBindingUtil;
import androidx.databinding.ViewDataBinding;

public abstract class BaseBindingActivity<T extends ViewDataBinding> extends BaseActivity {
    /************************************************************
     *  Static & Global Members
     ************************************************************/


    /************************************************************
     *  UI controls & Data members
     ************************************************************/
    public T binding;


    /************************************************************
     *  Overrides
     ************************************************************/
    @Override
    public void setContentView(int layoutResID) {
        this.binding = DataBindingUtil.inflate(getLayoutInflater(), layoutResID, null, true);
        binding.setLifecycleOwner(this);
        super.setContentView(this.binding.getRoot());
    }

    @Override
    protected void onResume() {
        super.onResume();

        // updateLang
        binding.invalidateAll();
    }

    /************************************************************
     *  Event Handler
     ************************************************************/


    /************************************************************
     *  Helpers
     ************************************************************/
    protected void initView() {
        binding.setLifecycleOwner(this);
    }

    /************************************************************
     *  Sub Classes
     ************************************************************/

}

File: app/src/main/java/com/victoria/bleled/base/BaseBindingActivity.kt
package com.victoria.bleled.base

import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding

abstract class BaseBindingActivity<T : ViewDataBinding> : BaseActivity() {
    /************************************************************
     * Static & Global Members
     */
    /************************************************************
     * UI controls & Data members
     */
    lateinit var binding: T

    /************************************************************
     * Overrides
     */
    override fun setContentView(layoutResID: Int) {
        binding = DataBindingUtil.inflate(layoutInflater, layoutResID, null, true)
        binding.lifecycleOwner = this
        super.setContentView(binding.root)
    }

    override fun onResume() {
        super.onResume()

        // update Language
        binding.invalidateAll()
    }
    /************************************************************
     * Event Handler
     */
    /************************************************************
     * Helpers
     */
    protected open fun initView() {
        binding.lifecycleOwner = this
    }
    /************************************************************
     * Sub Classes
     */
}